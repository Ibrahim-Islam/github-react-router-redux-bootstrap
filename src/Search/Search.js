import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { Button, Form, Input } from 'reactstrap';
import './Search.css';

class Search extends Component {

  constructor(props){
    super(props)
    this._handleFindClick = this._handleFindClick.bind(this);
  }

  _handleFindClick(event) {
    const usernameEl = document.getElementById('username');
    if (!usernameEl.checkValidity()) {
      document.querySelector('form').reportValidity();
      return;
    }
    this.props.history.push(`/${usernameEl.value}/repositories`);
  }

  render() {
    return (
      <div className="gh-form-body">
        <div className="container">
          <h1>Find public repositories</h1>
          <Form className="gh-search-form">
            <Input id="username" type="text" required placeholder="Enter A Github Username e.g. facebook" />
            <Button onClick={this._handleFindClick} color="primary">Find</Button>
          </Form>
        </div>
      </div>
    );
  }
}

export default withRouter(Search);
