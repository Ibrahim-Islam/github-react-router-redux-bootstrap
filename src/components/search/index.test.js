import React from 'react'
import ReactDOM from 'react-dom'
import Search from './index'
import { configure, shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

configure({ adapter : new Adapter() })

describe('<Search />', () => {

  it('should render without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Search />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('should match snapshot', () => {
    const wrapper = shallow(<Search />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should not submit form when enter key pressed inside username input', () => {
    const wrapper = shallow(<Search />)
    const event = { preventDefault: jest.fn(), stopPropagation: jest.fn() }
    const form = wrapper.find('Form').first()
    form.simulate('submit', event)
    expect(event.preventDefault).toHaveBeenCalledTimes(1)
  })

  it('should route to repositories page', () => {
    const historyMock = { push : jest.fn() }
    const wrapper = shallow(<Search history={historyMock} />)
    const button = wrapper.find('Button').first()
    wrapper.setState({ 
      value: 'test-user',
      valid: true
    })
    button.simulate('click')
    expect(historyMock.push.mock.calls[0][0]).toBe('/test-user/repositories')
  });

  it('should not route to repositories page if username empty', () => {
    const historyMock = { push : jest.fn() }
    const wrapper = shallow(<Search history={historyMock} />)
    const button = wrapper.find('Button').first()
    wrapper.setState({ 
      value: '',
      valid: true
    })
    button.simulate('click')
    expect(wrapper.find('.gh-error').length).toBe(1)
    expect(historyMock.push.mock.calls.length).toBe(0)
  });

});


//test pressing find with text in textbox will navigate away from page