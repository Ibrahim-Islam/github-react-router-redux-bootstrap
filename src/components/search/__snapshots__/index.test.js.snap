// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should match snapshot 1`] = `
<div
  className="gh-form-body"
>
  <div
    className="container"
  >
    <h1>
      Find public repositories
    </h1>
    <Form
      className="gh-search-form"
      onSubmit={[Function]}
      tag="form"
    >
      <Input
        className=""
        onChange={[Function]}
        placeholder="Enter A Github Username"
        type="text"
      />
      <Button
        color="primary"
        onClick={[Function]}
        tag="button"
      >
        Find
      </Button>
    </Form>
  </div>
</div>
`;
